### PR TITLE
add collections api definition

### PR DESCRIPTION
**Pull request by philvarner**
**Related Issue(s):** #134

**Proposed Changes:**

1. Add Collections conformance class defintion

**PR Checklist:**

- [X] This PR is made against the dev branch (all proposed changes except releases should be against dev, not master).
- [X] This PR has **no** breaking changes.
- [X] This PR does not make any changes to the core spec in the `stac-spec` directory (these are included as a subtree and should be updated directly in [radiantearth/stac-spec](https://github.com/radiantearth/stac-spec))
- [X] I have added my changes to the [CHANGELOG](https://github.com/radiantearth/stac-api-spec/blob/dev/CHANGELOG.md) **or** a CHANGELOG entry is not required.

### CODE REVIEW

File: collections/README.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
+# STAC API - Collections
+
+- [STAC API - Collections](#stac-api---collections)
+	- [Link Relations](#link-relations)
+	- [Endpoints](#endpoints)
+	- [Example](#example)
+	- [Conformance](#conformance)
+
+- **OpenAPI specification:** Missing
+- **Conformance URI:** <http://stacspec.org/spec/api/1.0.0-beta.1/extensions/collections>
```
Comment (matthewhanson) [inline collections/README.md:10]: should be 1.0.0-beta.2

File: collections/README.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+**NOTE**: *This conformance class is directly based on the [Features Collection](http://docs.opengeospatial.org/is/17-069r3/17-069r3.html#_collections_)
+section of OAFeat, which is in the process of becoming a 'building block in [OGC API - Common - Part 2: Geospatial 
+Data](http://docs.opengeospatial.org/DRAFTS/20-024.html) as the [Collections requirements 
+class](http://docs.opengeospatial.org/DRAFTS/20-024.html#rc_collections-section). Once the Common version is released we will 
+aim to align with it. But it still seems to be in flux.*
+
+## Link Relations
+
+The following Link relations should exist in the Landing Page (root).
```
Comment (matthewhanson) [inline collections/README.md:26]: Was this intended to have Link relations for the Landing page here on the Collections endpoint?

File: collections/README.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+			"links": [
+				{
+					"rel": "self",
+					"type": "application/json",
+					"href": "https://myservice.com/collections/cool-data"
+				},
+			],
+		}
+	],
+	"links": [
```
Comment (matthewhanson) [inline collections/README.md:81]: So no other links required: root, parent?

**Comment by matthewhanson** [comment]
This seems fine, but wondering what is it that isn't captured in the existing `http://www.opengis.net/spec/ogcapi-features-1/1.0/conf/core` conformance class for the /collections endpoint that requires a STAC specific Collections conformance class?

File: collections/README.md (hunk 2 above)
Comment (philvarner) [reply collections/README.md:26]: yes. We probably also need to clarify which link relations should exist on the /collections and /collections/{collectionid} endpoints for both this and ogcapi-features.

File: collections/README.md (hunk 3 above)
Comment (philvarner) [reply collections/README.md:81]: I'll improve this example.

**Comment by m-mohr** [comment]
> This seems fine, but wondering what is it that isn't captured in the existing `http://www.opengis.net/spec/ogcapi-features-1/1.0/conf/core` conformance class for the /collections endpoint that requires a STAC specific Collections conformance class?

It indicated that the collections are STAC compliant...

**Review by matthewhanson — APPROVED** [review]

**Comment by m-mohr** [comment]
Interesting PR, would like to review it as it's what we pioneered in the openEO API, but will likely take me some days to go through it.

**Comment by philvarner** [comment]
@matthewhanson This was Chris's idea, and it's basically to allow a user to advertise STAC-compliant Collections without needing to implement all of OGC API Features, e.g., they only want to implement Collections and Item Search. Ideally, there would be a way to define this without duplicating it from OAFeat, but this is what we have for now.

**Pull request merged**